flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestEnvironmentImpl.java

```diff
@@ -125,7 +126,13 @@ public void prepare(Config config) throws Exception {
         zookeeper = null;
         brokers.clear();
 
+        // Use a file lock to avoid race condition when creating and starting Zookeeper
+        // TestingServer, which may lead to java.net.BindException: Address already in use.
+        FileLock fileLock = new FileLock(getClass().getName() + "fileLock");
```

Comment:
I think the FileLock should be moved into `NetUtils#getAvailablePort` as this is the actual source of the port. As it's now, other test that use that function are circumventing your lock and let the job fail with the same error. Ideally, you can temporarily change `getAvailablePort` to return a constant port and the tests should still pass.

Reply:
Hi @AHeise ,
I moved the FlieLock to flink core within the same package as NetUtils and created a class named Port, which holds a port and FileLock. The FileLock would be automatically unlocked with the help of AutoClosable.